### PR TITLE
Update LuckPerms providers to use the latest API version

### PR DESCRIPTION
**Pull request by lucko**
Updates the LP providers to use the latest API version.

No functional change, but makes use of a method to retrieve a `Contexts` instance using the player object (faster), and removes the reliance on my Nexus repository.

### CODE REVIEW

File: bungee/luckperms/build.gradle [hunk 1]
```diff
@@ -16,11 +16,9 @@
  */
 
 repositories {
-    maven {
-        url "https://repo.lucko.me/"
-    }
+    mavenCentral()
```
Comment (CodeCrafter47) [inline bungee/luckperms/build.gradle]: No need for this one, maven central is configured in the root build.gradle https://github.com/CodeCrafter47/minecraft-data-api/blob/master/build.gradle#L42

**Review by CodeCrafter47 — COMMENTED** [review]
Does this break compatibility with older versions of LuckPerms?

Not specifically to this PR but generally I noticed that `/lp group <group> meta set weight <weight>` seems to have a different effect than `/lp group <group> setweight <weight>` which is kind of confusing to users trying to use the player order feature of BungeeTabListPlus.

**Comment by fnutt** [comment]
@CodeCrafter47 The current implementation returns errors in BungeeCord console - LOTS of them, all the time. This PR has fixed the issue for me.

`16:54:10 [WARNING] [BungeeTabListPlus] Failed to acquire data DataKey(id=luckperms:prefix, scope=Scope(id=bungee:player), type=de.codecrafter47.data.api.TypeToken@94e51e8, parameter=null) from <playername>
java.lang.NoSuchMethodError: me.lucko.luckperms.api.User.getUserDataCache()Ljava/util/Optional;
	at de.codecrafter47.data.bungee.luckperms.LuckPermsPrefixProvider.apply(LuckPermsPrefixProvider.java:43)
	at de.codecrafter47.data.bungee.luckperms.LuckPermsPrefixProvider.apply(LuckPermsPrefixProvider.java:29)
	at de.codecrafter47.data.api.AbstractDataAccess.lambda$addProvider$0(AbstractDataAccess.java:30)
	at de.codecrafter47.data.api.AbstractDataAccess.get(AbstractDataAccess.java:44)
	at de.codecrafter47.data.bungee.AbstractBungeeDataAccess.get(AbstractBungeeDataAccess.java:42)
	at de.codecrafter47.data.api.JoinedDataAccess.get(JoinedDataAccess.java:40)
	at codecrafter47.bungeetablistplus.managers.DataManager.updateData(DataManager.java:71)
	at net.md_5.bungee.scheduler.BungeeTask.run(BungeeTask.java:63)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)`

**Comment by fnutt** [comment]
> Not specifically to this PR but generally I noticed that /lp group <group> meta set weight <weight> seems to have a different effect than /lp group <group> setweight <weight> which is kind of confusing to users trying to use the player order feature of BungeeTabListPlus.

I agree about this. Struggling with this myself, after I moved from BungeePerms to LuckPerms. LuckPerms counterpart placeholders does not work as intended. I've raised this issue with @lucko before. Would very much like a fix on this, please.

**Comment by lucko** [comment]
@christer88 This PR doesn't address those issues - they have already been fixed - but the fixed version doesn't seem to be on SpigotMC yet.

> Does this break compatibility with older versions of LuckPerms?

Yes - it uses API features which weren't in previous releases - however BTLP is currently depending on a version several months out of date.

I appreciate you've had to fix compatibility once before (since my original PR for support) - the plugin is still evolving (albeit now at a much slower rate) - such drastic backwards incompatible changes hopefully won't happen again.

**Comment by lucko** [comment]
> Not specifically to this PR but generally I noticed that /lp group <group> meta set weight <weight> seems to have a different effect than /lp group <group> setweight <weight> which is kind of confusing to users trying to use the player order feature of BungeeTabListPlus.

Using the `meta set` command is not the correct way to define weight - the value is however exposed via the API as a meta value, which is perhaps what gave you that impression. End users have no reason to think that though.

https://github.com/lucko/LuckPerms/blob/master/common/src/main/java/me/lucko/luckperms/common/caching/type/MetaAccumulator.java#L103

File: bungee/luckperms/build.gradle (hunk 1 above)
Comment (lucko) [reply bungee/luckperms/build.gradle]: Fixed

**Comment by fnutt** [comment]
@lucko 

> playerOrder: "bungeePermsGroupInfo,playerServerFirst,serverAlphabetically,playerWorldFirst,worldName,youFirst,AFKLast,alphabetically"

This worked before with BungeePerms, but using Luckperms and 'LuckPermsGroupInfo' does not. There's definently something not quite right, as @CodeCrafter47 pointed out.

Ref: https://www.spigotmc.org/resources/bungeetablistplus.313/update?update=146411

**Pull request merged**